### DISCUSSION TITLE
Voice Intents "store" (like Add-ons)

### DISCUSSION THREAD

**Discussion by Nerivec** [home-assistant/frontend#19012] [Configuration]
With voice making its way into Home Assistant, we need a better way to handle loading/unloading intents into Home Assistant.

I see a system similar to Add-ons being the most "convenient". Users could check/uncheck intents in a curated list and that would automatically add/remove support on their machine.

- Support for creation/linking of helpers/automations/scripts (similar to how the Energy dashboard creates/links forecast when adding solar power), since most intents would use at least one entity.
- Support for misc entities; right now, "included" intents are very limited, mostly device-specific, need to add things like time, weather, etc. Some, like timer management, would probably require new integrations for better handling (to avoid cheap hacks like we have to do atm). This is only loosely frontend-related, but it is a major obstacle for adoption so I had to mention it.

A per-entity support for speech output customization (inside entity settings>voice assistants) would be nice; a simple template field to override "default" would do, with perhaps support like sentences, to provide easy randomness (that last one goes for every TTS template, example below).

```yaml
(It's|Temperature is) {value}{unit} outside.
```

**Comment by Hedda** [comment]
@Nerivec tip if you are thinking about implementing this then post to -> https://github.com/home-assistant/architecture/discussions
